src/test/java/de/komoot/photon/ApiIntegrationTest.java

```diff
@@ -72,6 +73,57 @@ void testCorsOriginIsSetToSpecificDomain() throws Exception {
         assertEquals("www.poole.ch", connection.getHeaderField("Access-Control-Allow-Origin"));
     }
 
+
+    /*
+     * Test that the Access-Control-Allow-Origin header is set to the matching domain
+     */
+    @Test
+    void testCorsOriginIsSetToMatchingDomain() throws Exception {
+        App.main(new String[]{"-cluster", TEST_CLUSTER_NAME, "-listen-port", Integer.toString(LISTEN_PORT), "-transport-addresses", "127.0.0.1",
+                "-cors-origin", "www.poole.ch,alt.poole.ch"});
+        awaitInitialization();
+
+        try {
+            // allows us to set the Origin header
+            System.setProperty("sun.net.http.allowRestrictedHeaders", "true");
+
+            String[] origins = {"www.poole.ch", "alt.poole.ch"};
+            for (String origin: origins) {
+                URLConnection urlConnection = new URL("http://127.0.0.1:" + port() + "/api?q=berlin").openConnection();
+                urlConnection.setRequestProperty("Origin", origin);
+                HttpURLConnection connection = (HttpURLConnection) urlConnection;
+                assertEquals(origin, connection.getHeaderField("Access-Control-Allow-Origin"));
+            }
+        } finally {
+            System.setProperty("sun.net.http.allowRestrictedHeaders", "false");
+        }
+    }
+
+    /*
+     * Test that the Access-Control-Allow-Origin header does not return mismatching origins
+     */
+    @Test
+    void testMismatchedCorsOriginsAreBlock() throws Exception {
+        App.main(new String[]{"-cluster", TEST_CLUSTER_NAME, "-listen-port", Integer.toString(LISTEN_PORT), "-transport-addresses", "127.0.0.1",
+                "-cors-origin", "www.poole.ch,alt.poole.ch"});
+        awaitInitialization();
+
+        try {
+            // allows us to set the Origin header
+            System.setProperty("sun.net.http.allowRestrictedHeaders", "true");
+
+            String[] origins = {"www.randomsite.com", "www.arbitrary.com"};
+            for (String origin: origins) {
+                URLConnection urlConnection = new URL("http://127.0.0.1:" + port() + "/api?q=berlin").openConnection();
+                urlConnection.setRequestProperty("Origin", origin);
+                HttpURLConnection connection = (HttpURLConnection) urlConnection;
+                assertNotEquals(origin, connection.getHeaderField("Access-Control-Allow-Origin"));
```

Comment:
If I read the code right, this should return the first configured origin. So better test for `assertEquals("www.poole.ch", ...`.

Reply:
Thanks for the feedback.

I've made this change, and fixed the tests with

```
test {
    systemProperty "sun.net.http.allowRestrictedHeaders", "true"
}
```